pkg/hooks/manager.go

```diff
@@ -0,0 +1,228 @@
+	if err != nil {
+		return err
+	}
+	log.Log.Info("Collected all requested hook sidecar sockets")
+
+	sortCallbacksPerHookPoint(callbacksPerHookPoint)
+	log.Log.Infof("Sorted all collected sidecar sockets per hook point based on their priority and name: %v", callbacksPerHookPoint)
+
+	m.callbacksPerHookPoint = callbacksPerHookPoint
+
+	return nil
+}
+
+// TODO: Handle sockets in parallel, when a socket appears, run a goroutine trying to read Info from it
+func collectSideCarSockets(numberOfRequestedHookSidecars uint, timeout time.Duration) (map[string][]*callackClient, error) {
+	callbacksPerHookPoint := make(map[string][]*callackClient)
+	processedSockets := make(map[string]bool)
+
+	timeoutCh := time.After(timeout)
+
+	for uint(len(processedSockets)) < numberOfRequestedHookSidecars {
+		sockets, err := ioutil.ReadDir(HookSocketsSharedDirectory)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, socket := range sockets {
+			select {
+			case <-timeoutCh:
+				return nil, fmt.Errorf("Failed to collect all expected sidecar hook sockets within given timeout")
+			default:
+				if _, processed := processedSockets[socket.Name()]; processed {
+					continue
+				}
+
+				callackClient, notReady, err := processSideCarSocket(HookSocketsSharedDirectory + "/" + socket.Name())
+				if notReady {
+					log.Log.Info("Sidecar server might not be ready yet, retrying in the next iteration")
+					continue
+				} else if err != nil {
+					log.Log.Reason(err).Infof("Failed to process sidecar socket: %s", socket.Name())
+					return nil, err
+				}
+
+				for _, subsribedHookPoint := range callackClient.subsribedHookPoints {
+					callbacksPerHookPoint[subsribedHookPoint.GetName()] = append(callbacksPerHookPoint[subsribedHookPoint.GetName()], callackClient)
+				}
+
+				processedSockets[socket.Name()] = true
+			}
+		}
+
+		time.Sleep(time.Second)
+	}
+
+	return callbacksPerHookPoint, nil
+}
+
+func processSideCarSocket(socketPath string) (*callackClient, bool, error) {
+	conn, err := dialSocket(socketPath)
+	if err != nil {
+		log.Log.Reason(err).Infof("Failed to Dial hook socket: %s", socketPath)
+		return nil, true, nil
+	}
+	defer conn.Close()
+
+	infoClient := hooksInfo.NewInfoClient(conn)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	info, err := infoClient.Info(ctx, &hooksInfo.InfoParams{})
+	if err != nil {
+		return nil, false, err
+	}
+
+	versionsSet := make(map[string]bool)
+	for _, version := range info.GetVersions() {
+		versionsSet[version] = true
+	}
+
+	if _, found := versionsSet[hooksV1alpha1.Version]; found {
+		return &callackClient{
+			SocketPath:          socketPath,
+			Version:             hooksV1alpha1.Version,
+			subsribedHookPoints: info.GetHookPoints(),
+		}, false, nil
+	} else {
+		return nil, false, fmt.Errorf("Hook sidecar does not expose a supported version. Exposed versions: %v, supported versions: %s", versionsSet, hooksV1alpha1.Version)
```

Comment:
It would make testing easier for 3rd party hook authors if the errors were in the info.go package so you can trigger the errors and make sure you get the right error for the expected error condition.

Reply:
But some of the messages are formatted, keeping it in a constant would not help much. Do you still think it will be worth it?